src/client_side_reply.cc

```diff
@@ -1484,17 +1485,18 @@ clientReplyContext::identifyStoreObject()
 {
     HttpRequest *r = http->request;
 
-    // client sent CC:no-cache or some other condition has been
-    // encountered which prevents delivering a public/cached object.
-    // XXX: The above text does not match the condition below. It might describe
-    // the opposite condition, but the condition itself should be adjusted
-    // (e.g., to honor flags.noCache in cache manager requests).
-    if (!r->flags.noCache || r->flags.internal) {
+    if (r->flags.noCache && !r->flags.internal) {
+        identifyFoundObject(nullptr, "no-cache"); // skip Store lookup for  "external" no-cache requests
+    } else if (ForSomeCacheManager(r->url.path())) { // cache manager requests ought to have flags.noCache
```

Comment:
The comment sounds misleading to me -- this condition/use-case is not about flags; is is about the fact that cache manager requests must not be satisfied from the cache. We are already documenting that below so we can just drop this comment. The condition itself is self-documented.

```suggestion
    } else if (ForSomeCacheManager(r->url.path())) {
```

Reply:
Removed.